homeassistant/components/recorder/purge.py

```diff
@@ -1,88 +1,50 @@
 """Purge old data helper."""
-from datetime import timedelta
+from datetime import datetime, timedelta
 import logging
 import time
+from typing import TYPE_CHECKING
 
 from sqlalchemy.exc import OperationalError, SQLAlchemyError
+from sqlalchemy.orm.session import Session
 
 import homeassistant.util.dt as dt_util
 
 from .models import Events, RecorderRuns, States
-from .util import execute, session_scope
+from .repack import repack_database
+from .util import session_scope
+
+if TYPE_CHECKING:
+    from . import Recorder
 
 _LOGGER = logging.getLogger(__name__)
 
+MAX_ROWS_TO_PURGE = 1000
+
 
-def purge_old_data(instance, purge_days: int, repack: bool) -> bool:
+def purge_old_data(instance: "Recorder", purge_days: int, repack: bool) -> bool:
```

Comment:
```suggestion
def purge_old_data(instance: Recorder, purge_days: int, repack: bool) -> bool:
```
You should be able to use `Recorder` (here and the other times) without quotes if you add
```py
from __future__ import annotations
```

Reply:
Thanks